app/controllers/foreman_tasks/api/recurring_logics_controller.rb

```diff
@@ -39,9 +45,22 @@ def resource_class
         ForemanTasks::RecurringLogic
       end
 
+      api :POST, '/recurring_logics/bulk_destroy', N_('Delete recurring logics by search query')
+      param :search, String, :desc => N_('Search query'), :required => true
+      def bulk_destroy
+        if params[:search].blank?
+          raise BadRequest, _('Please provide a search parameter in the request')
```

Comment:
How about using something like this instead of creating a custom exception class?
```ruby
payload = { :error => { :message => _("Required parameter 'search' missing in the request") } }
render :json => payload, :status => :bad_request
return
```

Reply:
Why not use `BadRequest`? it's used in tasks

Reply:
Ah, you're right, completely forgot about it. Nevermind then